server/notifications.go

```diff
@@ -5,91 +5,87 @@
 package server
 
 import (
-	"context"
 	"sync"
-
-	"github.com/gofrs/uuid"
-)
-
-const (
-	// how many notifications can be in a notification buffer.
-	notificationBufferSize = 16
+	"sync/atomic"
 )
 
-type change struct {
-	// persistedIndex is a changed persisted index or `nil` if the value has not changed.
-	persistedIndex *int64
+// state is an extendable struct that stores
+// the information subscribers need to be notified about
+type state struct {
+	persistedIndex uint64
 }
 
-// Any returns true if there is a change.
-func (c change) Any() bool {
-	return c.persistedIndex != nil
+// Equals returns true if 2 instances of the state are equal
+func (s *state) Equals(v *state) bool {
+	return (s == nil && v == nil) || (s != nil && v != nil && s.persistedIndex == v.persistedIndex)
 }
 
 type notifications struct {
-	subscribers map[uuid.UUID]chan change
-	mutex       *sync.Mutex
+	mutex *sync.RWMutex
```

Comment:
Why does this need to be a pointer? The zero value of a mutex is usable.

Reply:
Ah that would only apply if you you were intending users to make copies of `notifications`. Regardless it isn't incorrect, resolving.